hazelcast/src/main/java/com/hazelcast/map/impl/mapstore/BasicMapStoreContext.java

```diff
@@ -242,7 +247,7 @@ public void run() {
     private void selectOwnedKeys(Iterable loadedKeys, MapServiceContext mapServiceContext) {
 
         initialKeys.clear();
-        int maxSizePerNode = getApproximateMaxSize(getMaxSizePerNode()) - 1;
+        int maxSizePerNode = getApproximateMaxSize(maxSizeConfig, PER_NODE) - 1;
```

Comment:
I know this is not in the scope of this PR but can you make this `-1` logic more clear? Why do we have that? Seems not understandable at first look. Btw, assuming that is related to 0 is integer max.?